Clean up code to match PEP8

linusg opened the pull request:
This will fix Issue #93 by applying PEP8's rules to most of the code. I'm sorry if this introduces a bug, if so, let me know immediately.

This refactoring includes:
- Have two empty lines before each global function
- Use `'{0} {1}'.format(str1, str2)` instead of `str1 + ' ' + str2`
  Sometimes this will make lines longer, sometimes shorter.
- Start all comments with `# + space + comment`, so `# space in front` not `#nospacehere`
- Make lines not longer than 80 characters in most cases (OK, 80-100 are fine)
- Rename some variables to make more sense
- Add some missing code like returns and Exceptions
- Transform comments above functions into docstrings
- Remove some way too verbose comments
- Fix some errors introduced with the first code cleanup
    
Not yet done, but IMHO not that important for now:
- Rename all 'file' variables, only for Python 2
- Add params to docstrings
- Use OOP/clases to avoid globals. IMHO the codebase is big enough.

ritiek commented:
There were some encoding problems due to which the tests were failing but I've fixed them. Just to be on the safe side, I'll merge once Travis CI completes testing again. Thanks a lot again for helping out!

linusg commented:
Ah, damn. I tested under Linux, and only Linux. Python Unicode sucks. And really, is it just Windows?
Anyway, thanks for fixing this for me. Closing this for now :+1:

ritiek commented:
No, it is both linux and windows. It happens when the song name contains special characters like in this [test case](https://github.com/ritiek/spotify-downloader/blob/master/test/test_single.py). Yep, python 2 unicode sucks. But for some reason, windows thows even more unicode errors than linux/mac. Everything seems good now btw, I think we're ready to merge this!

linusg commented:
Reminder to myself: test using the _given_ tests, not by downloading a random song I like. Sorry for the extra work...

ritiek commented on spotdl.py:
```diff
@@ -74,8 +75,8 @@ def generate_youtube_url(raw_song):
             # confirm the video result is not an advertisement
-            if not x.find('channel') == -1 or not x.find('googleads') == -1:
-                print(str(check) + '. ' + x.get_text())
+            if x.find('channel') > -1 or x.find('googleads') > -1:
```
There is a problem in `--manual` mode too. We do not want `channel` or `googleads` words to be found but in your case it expects to find either one of them. Something like below should be better and more readable than my previous attempt.

`if not (x.find('channel') == -1 or x.find('googleads') == -1):`

linusg replied:
Yes, I simplified it. Guess you're wrong, `not x.find('channel') == -1` _is_
 the same as `x.find('channel') > -1`. Index -1 means not found, so yours says 'if one of channel or googleads found', whereas mine says 'if one of channel or googleads found' (when the index is 0 or greater, it was found).

Example:
```
>>> x = 'No channe_l or google_ads in here'  # Notice the _
>>> not x.find('channel') == -1 or not x.find('googleads') == -1
False
>>> x.find('channel') > -1 or x.find('googleads') > -1
False
>>> x = 'No channe_l in here but googleads'
>>> not x.find('channel') == -1 or not x.find('googleads') == -1
True
>>> x.find('channel') > -1 or x.find('googleads') > -1
True
>>> 
```

ritiek replied:
I am all confused now but for some reason `--manual` mode won't work with your snippet..

linusg replied:
I see. It must be something different though. Give me some time, I'll find it (hopefully). I had this strange feeling all the time that this much changes will break something :thinking:

ritiek replied:
Any update on this or should we fall back on `not (x.find('channel') == -1 or x.find('googleads') == -1)`?

ritiek commented on spotdl.py:
```diff
@@ -85,46 +86,52 @@ def generate_youtube_url(raw_song):
         # otherwise keep iterating until it is not
-        while not result.find('channel') == -1 or not result.find('googleads') == -1:
-            result = items_parse.find_all(attrs={'class': 'yt-uix-tile-link'})[check]['href']
+        while result.find('channel') > -1 or result.find('googleads') > -1:
```
You'll have to those changes here as well. 😄

linusg replied:
Yep, simplified this one as well :wink:

linusg commented:
Sorry, still no progress on this. Let me see if I can fix it today.

ritiek commented:
Its all good, take your time and lets see if we can come up to some elegant solution for `--manual` mode. :blush:

linusg commented:
<s>Sorry for the delay, I'm busy again. Try it, it's not this line. As I explained above, it's the same as before but simplified. The mistake is somewhere else :disappointed_relieved: 

Or, just ignore this whole issue and PR, that would be the most simple thing.</s>

Nope, I finally found it!!!!!!!!11!eleven!!111 (:wink:)

So, let me tell you:
- First of all, in [line 77](https://github.com/ritiek/spotify-downloader/blob/master/spotdl.py#L77) the variable `x` is of the type `<class 'bs4.element.Tag'>`! So using the `find` method will not return `-1` in case nothing was found but `None` (see [this StackOverflow answer](https://stackoverflow.com/a/25003439/5952681)).

So let's change

    if not x.find('channel') == -1 or not x.find('googleads') == -1:

to

    if x.find('channel') is None and x.find('googleads') is None:

You see, the logic was wrong (even before I changed it). You've checked if the result is either a channel or an ad, and if so add it to the link list?! `not (x.find('channel') == -1 or x.find('googleads') == -1)` would be wrong too, it would add a link to the list if both `'channel'` and `'googleads'` are contained. But please, if I tell completely wrong things, tell me. It's just Boolean Logic after all.

Next, 

    while not result.find('channel') == -1 or not result.find('googleads') == -1:

can be simplified to

    while result.find('channel') > -1 or result.find('googleads') > -1:

see my previous comment. Here, we have a string so no occurrence will return `-1`.

I also fixed some minor things like wrong braces or variable names I forgot during the refactoring. I ran the test, for reference:

```
linus@sony-vaio ~/Code/python/projects/spotify-downloader $ python3 -m pytest test
=============================================================== test session starts ================================================================
platform linux -- Python 3.5.2, pytest-3.1.3, py-1.4.34, pluggy-0.4.0
rootdir: /home/linus/Code/python/projects/spotify-downloader, inifile:
collected 10 items 

test/test_single.py .......
test/test_username.py ...

============================================================ 10 passed in 31.36 seconds ============================================================
linus@sony-vaio ~/Code/python/projects/spotify-downloader $ 
```

_But_ only after deleting the file `list.txt`! You may add a note about this in the README. After running them one time, they'll fail the next time, because the list is growing.

```
>       assert (expect_lines == lines and expect_song == song)
E       assert (14 == 27)

test_username.py:57: AssertionError
```

Last but not least, by changing the return codes of the convert functions from `None` to `0`, you broke the test. I fixed it, though :wink: 

Please don't see this as critic, just want to be constructive. Don't hesitate asking further questions or telling me _I'm_ wrong. After all I hope this is finally fixed :grin:

ritiek commented:
I get it now. So, it was `BeautifulSoup.find()` in action instead of `string.find()`. Everything makes sense now! Did not think about that earlier.

I'll update the test to overwrite `list.txt` (instead of appending) soon. I think I am also going to write more tests, the present ones do not cover some other edge cases.

Thanks a lot for your contribution! I'll test your PR with a few more cases, then it should be finally be merged. 😄

linusg commented:
Phew...
OK, let me know when you're done. Thanks for your patience :smile:

linusg commented:
Thank you again!
